security/pkg/pki/util/keycertbundle.go

```diff
@@ -253,19 +253,19 @@ func (b *KeyCertBundle) UpdateVerifiedKeyCertBundleFromFile(certFile string, pri
 	return nil
 }
 
-// ExtractRootCertExpiryTimestamp returns the unix timestamp when the root becomes expires.
-func (b *KeyCertBundle) ExtractRootCertExpiryTimestamp() (float64, error) {
+// ExtractRootCertExpiryTimestamp returns a time.Time of when the root cert expires
+func (b *KeyCertBundle) ExtractRootCertExpiryTimestamp() (*time.Time, error) {
 	return extractCertExpiryTimestamp("root cert", b.GetRootCertPem())
```

Comment:
I think if we are touching those methods and updating docs - we should also clarify what they do.

The root certs is a list - not a single certificate, even in the case of the self-signed root it is possible to have the ca pem updated by a tool ( to support rotation or multiple clusters). This method only looks at the expiration of the first cert, which may or may not be the root that signed the current cert. 

The second method only looks at the leaf certificate - does not consider the expiration of any intermediate cert. Normally a well behaved CA should not issue a cert that expires after the intermediates or root, and istio certs expire relatively fast - so the expiry of the leaf is ok.

Reply:
What do you think about returning the timestamp/expiry seconds for the root cert that will expire soonest? As the aim of this metric is to alert the user to take action for certificates at risk of expiring.

Reply:
Not sure - the normal procedure is to create a new root cert, add it to the bundle - but keep the old root around for the workloads that still use it. 

It is fine to have an expired or about the expire root in the bundle - the alert is not about a root about to expire, but about certificate chains that are anchored to a root that is close to expiration. If all the cert chains are migrated to the new root - there is nothing to alert and no reason to have false alerts or remove the old root prematurely ( who knows what VM still has a cert on the old root ) ? I assume the alert will be few days or weeks before the root is expired.